### DISCUSSION TITLE
control_c768 RT test

### DISCUSSION THREAD

**Discussion by RuiyuSun** [ufs-community/ufs-weather-model#2291] [Q&A]
Have anyone used control_c768 on Hera?

**Comment by gspetro-NOAA** [comment]
Hi @RuiyuSun ,

The regression tests are run regularly on Hera whenever new code is ready to be merged. The `control_c768` test is run as part of that process. Do you have a specific question about the test?

Best,
Gillian

**Reply by RuiyuSun** [reply]
Thank @gspetro-NOAA for answering my question!  

The test is not in the rt.conf. I added and used the following command to run the test on Hera. The run failed. I am redoing it  to reproduce the error. 

 ./rt.sh -a fv3-cpu -k -n "control_c768 intel" 

Is this not the way to run the test?

**Reply by gspetro-NOAA** [reply]
Ah, my mistake! It is not run every time PRs are merged because it is a more costly test. It is, however, part of the `rt_weekly.conf` [test suite](https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/rt_weekly.conf) and is therefore run when that is run. You can find some information on running the `control_c768` test (or any test) in [this section](https://ufs-weather-model.readthedocs.io/en/develop/BuildingAndRunning.html#running-the-model) of the WM User's Guide. The command would indeed be: 
```
./rt.sh -a fv3-cpu -k -n control_c768 intel
```
If that doesn't work, let me know, and I'll see if I can get one of our code managers to work with you.

**Reply by BrianCurtis-NOAA** [reply]
-n would need it in quotes "control_c768 intel"

**Reply by gspetro-NOAA** [reply]
@BrianCurtis-NOAA Thanks for chiming in! The test wasn't running with it in quotes. Do you have suggestions about what might be wrong instead? It seems the command is not the problem. @RuiyuSun It would be helpful to have more information about what errors you are seeing so that we can help you troubleshoot.

**Reply by RuiyuSun** [reply]
I just did at test and it failed with the following error in the run log file at /scratch1/NCEPDEV/global/Ruiyu.Sun/save/git/GFSv16_GFDLMPv1_bug_atfirststep/ufs-weather-model/tests/logs/log_hera . 

++ '[' .false. == .true. ']'
++ cp '/scratch2/NAGAPE/epic/UFS-WM_RT/NEMSfv3gfs/input-data-20221101/FV3_fix_tiled/C768/C768*.nc' .
cp: cannot stat '/scratch2/NAGAPE/epic/UFS-WM_RT/NEMSfv3gfs/input-data-20221101/FV3_fix_tiled/C768/C768*.nc': No such file or directory
+ '[' 1 -eq 0 ']'
+ write_fail_test
+ echo 'control_c768_intel failed in run_test'
+ exit 1

Please see below for screen output: 

Linking /scratch1/NCEPDEV/stmp2/Ruiyu.Sun/FV3_RT/rt_1931741 to /scratch1/NCEPDEV/global/Ruiyu.Sun/save/git/GFSv16_GFDLMPv1_bug_atfirststep/ufs-weather-model/tests/run_dir
Run regression test in: /scratch1/NCEPDEV/stmp2/Ruiyu.Sun/FV3_RT/rt_1931741
rt.sh: Checking & Updating test configuration...
rt.sh: Running compile atm_dyn32_intel
rt.sh: Compile atm_dyn32_intel completed.
rt_utils.sh: control_c768_intel: Computing PET bounds and tasks.
rt.sh: Running test control_c768_intel using compile atm_dyn32_intel
rt.sh: Getting error information...
Exited at line 1388 having code 1
rt.sh: Exited abnormally, killing workflow and cleaning up
rt.sh: Cleaning up...
rt.sh: Exiting.

**Reply by BrianCurtis-NOAA** [reply]
@gspetro-NOAA Thanks for the clarification, I should have read up on the prior discussion a bit more. The weekly tests are not run, as far as I know. So there's a high chance that this run failure is due to a c768 configuration not getting the testing it needs. I also do not know if the c768 inputs have been updated as needed.

As stated, If there's any output that you do not understand, then feel free to post it here and hopefully you can get some direction on where to go.

**Reply by BrianCurtis-NOAA** [reply]
@RuiyuSun It does seem odd we have a C768 test without having the tiled fix files for it. My guess is we stopped supporting that resolution in regression testing a while back.

There are workflows available that can generate a set of tiled fix files for you at that resolution if you need them, though.

**Reply by RuiyuSun** [reply]
@BrianCurtis-NOAA I see. Thanks.

**Reply by gspetro-NOAA** [reply]
@RuiyuSun I got confirmation from @jkbk2004 that we no longer support this test, so the ICs won't be in the default location. However, you can generate the ICs yourself using [GFS data](https://registry.opendata.aws/noaa-gfs-bdp-pds/) (it takes a moment to load when you click Browse Files) and the UFS_UTILS [chgres_cube utility](https://noaa-emcufs-utils.readthedocs.io/en/latest/ufs_utils.html#chgres-cube) to change the resolution. The [WM wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Getting-Started#4-run-the-test-case) has some information on how to run a sample case, and it notes that you may need to modify file-related variables in `rt.sh` (e.g., `dprefix`, `DISKNM`) if you are staging the data in a non-default location.

**Reply by RuiyuSun** [reply]
Thank you @gspetro-NOAA for the instruction!
